Tests/SonarScanner.MSBuild.Shim.Test/PropertiesFileGeneratorTests.cs

```diff
@@ -418,15 +455,55 @@ public void FileGen_FilesOutsideProjectPath()
             var result = new PropertiesFileGenerator(config, logger).GenerateFile();
 
             // Assert
-            AssertExpectedStatus("projectName", ProjectInfoValidity.NoFilesToAnalyze, result);
+            AssertExpectedStatus("project", ProjectInfoValidity.NoFilesToAnalyze, result);
             AssertExpectedProjectCount(1, result);
-
             // No files -> project file not created
             AssertFailedToCreatePropertiesFiles(result, logger);
+            logger.AssertWarningsLogged(1);
+            logger.AssertSingleWarningExists($"File '{fileOutsideRootPath}' is not located under the root directory");
         }
 
         [TestMethod]
-        public void FileGen_SharedFiles()
+        public void GenerateFile_FilesOutOfProjectPath_PrintsCorrectWarnings()
+        {
+            // Files outside the project root should be ignored
+
+            // Arrange
+            var testDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var nugetCacheDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, new string[] { ".nuget", "packages" });
+
+            var projectDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, "project");
+            var projectPath = Path.Combine(projectDir, "project.proj");
+            var projectInfo = TestUtils.CreateProjectInfoInSubDir(testDir, "project", null, Guid.NewGuid(), ProjectType.Product, false, "UTF-8", projectPath); // not excluded
+
+            string[] files = { "dllFile.dll", "exeFile.exe", "txtFile.txt" };
+            var filesToBeAnalyzedPaths = new List<string>();
+            foreach (var fileName in files)
+            {
+                filesToBeAnalyzedPaths.Add(TestUtils.CreateEmptyFile(TestContext.TestDir, fileName));
+            }
+            filesToBeAnalyzedPaths.Add(TestUtils.CreateEmptyFile(nugetCacheDir, "FileInNugetCache.cs"));
+
+            // To add the files above, to the list of files that are to be analyzed, you need to add their paths to
+            // the "contentList.txt" which is placed inside the projectDir folder.
+            var contentFileListPath = TestUtils.CreateFile(projectDir, "contentList.txt", string.Join(Environment.NewLine, filesToBeAnalyzedPaths));
+            // Add the file path of "contentList.txt" to the projectInfo.xml
+            TestUtils.AddAnalysisResult(projectInfo, AnalysisType.FilesToAnalyze, contentFileListPath);
+
+            var logger = new TestLogger();
+            var config = CreateValidConfig(testDir);
+
+            // Act
+            var result = new PropertiesFileGenerator(config, logger).GenerateFile();
```

Comment:
can you also add `.EXE` and `.DLL` files inside the base dir and verify they don't get added in the properties ?